packages/graphback-datasync/src/providers/DatasyncMongoDBDataProvider.ts

```diff
@@ -22,45 +23,156 @@ export class DataSyncMongoDBDataProvider<Type = any> extends MongoDBDataProvider
       throw e;
     });
     this.coerceTSFields = true;
+    this.conflictEngine = conflictEngine;
   }
 
   public async create(data: any, context: GraphbackContext): Promise<Type> {
     data._deleted = false;
+    if (this.conflictEngine !== undefined) {
+      const { next, conflictFieldName } = this.conflictEngine;
+      data[conflictFieldName] = next(undefined, data);
+    }
 
     return super.create(data, context);
   }
 
   public async update(data: any, context: GraphbackContext): Promise<Type> {
-    const conflict = await this.checkForConflicts(data, context);
 
-    if (conflict !== undefined) {
-      throw new ConflictError(conflict);
+    const { idField } = getDatabaseArguments(this.tableMap, data);
+
+    if (!idField.value) {
+      throw new NoDataError(`Cannot update ${this.collectionName} - missing ID field`)
     }
 
-    // TODO use findOneAndUpdate to check consistency afterwards
-    return super.update(data, context);
+    const objectId = new ObjectId(idField.value);
```

Comment:
I really struggle to understand what is happening here.

Do we want to maintain single Provider using conflicts and without conflicts? Would it make sense to extend it for better code/visiblity etc.

Reply:
I was under the impression that datasync = delta, and conflict would be opt-in, so wondering if that was a misunderstanding

Reply:
Yep.. but Initially seen that as two separate providers. 
WHY - entire delta table and tons of other complex elements are simply not needed if you just want to have delta. Soft deletes approach can be reused as it has very limited impact. This means that conflict can be addon on top of very robust API we have.

This will mean that testing will be simpler and we will avoid complexity, tons of ifs all over the provider that looks very very well done on master - and IMHO should stay this way.

So TL;DR - user API should be making decision by picking different data provider rather than dropping all into datasync provider. 

This will also make it really easy to implement schema markers. 
Just datasync is needed and you control conflicts by picking different helper. 
I might be totally wrong but when I think about using this myself - this approach will make a lot of sense. What was proposed here brings mental challenge to me to understand what will happen. 

What happens if I do not provide conflict? What is versioned conflict? Should I provide my own. tons of info and reading required upfront. 

What we really want is just.. Hey .. want conflict - use this method. If not stay with this stuff that is on master.

Reply:
This could be on me somewhat..

@ssd71 originally had separate providers - I saw that there was very little difference between the providers, only the extra code for conflict handling and thought it would reduce API complexity by having to mix multiple providers with very little difference. 

https://github.com/aerogear/graphback/pull/1647#discussion_r449454898

Reply:
It's still two providers though, https://github.com/aerogear/graphback/pull/1755/files/526dbcf47e7714add3a74ca55f88e8aed592f6db#diff-7cfe2dc3aac0bbf8cc840e0fab566a2f
and https://github.com/aerogear/graphback/pull/1755/files/526dbcf47e7714add3a74ca55f88e8aed592f6db#diff-be2392f5aaf3d58e02d7e9c6429f279c
Sorry about the confusion